jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -441,7 +441,10 @@ private Runnable onFailure(Throwable x, boolean remote)
             // If not handled, then we just fail the request callback
             if (!_handled && _handling == null)
             {
-                task = () -> _request._callback.failed(x);
+                if (LOG.isDebugEnabled())
+                    LOG.debug("failing request not yet handled {} {}", _request, this);
+                Callback callback = _request._callback;
+                task = () -> callback.failed(x);
```

Comment:
This does not feel right.

`_request` is nulled out by `recycle()`.

How it is possible that `recycle()` is called _before_ one of the 3 legs of the stool (the `Handler` callback) is complete?

@gregw @lorban thoughts?